src/puppetlabs/trapperkeeper/authorization/rules.clj

```diff
@@ -170,7 +163,8 @@
   [rules :- Rules
    request :- ring/Request
    name :- schema/Str]
-  (if-let [ { matched-rule :rule matches :matches } (some #(match? % request) rules)]
+  (if-let [{matched-rule :rule matches :matches}
+           (some #(match? % request) (sort-rules rules))]
```

Comment:
One other thought I'd had on this is that we might consider having the initial construct that the rules are loaded into be a Clojure `sorted-set` rather than just a list/set and have the sorting be done implicitly as rules are added to the structure - similar to what we're already doing with the ACE (access control entries) that comprise the sorted ACL (access control list).  Could even do the validation for duplicates as part of the comparator so that we don't end up having to walk the list multiple times and question later on whether the validation has been completely done or not by the time we get to an arbitrary point in execution.  If we did this, we could have schema validate that the structure type implements `Sorted`, so we'd know that the sorting had to have been done on the input argument.  Not sure if the extra turmoil to go with something like this would be justified, though, so I'm not suggesting we necessarily add that to this PR.